Source/CMakeLists.txt

```diff
@@ -13,6 +13,19 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
+  add_definitions(/D _CRT_NONSTDC_NO_WARNINGS)
+  add_definitions(/utf-8)
+
+  # Use PCH
```

Comment:
This seems a little hacky. Are there any cmake functions which can set up the precompiled header without resorting to hardcoding compiler-specific parameters? What about non-windows?

Reply:
Not possible any other way using cmake only afaik.